flink-python/pyflink/fn_execution/datastream/window/window_assigner.py

```diff
@@ -0,0 +1,376 @@
+
+
+class CountTrigger(Trigger[object, CountWindow]):
+    """
+    A Trigger that fires once the count of elements in a pane reaches the given count.
+    """
+    def __init__(self, window_size: int):
+        self._window_size = int(window_size)
+        self._count_state_descriptor = ReducingStateDescriptor(
+            "trigger_counter", lambda a, b: a + b, Types.BIG_INT())
+
+    def on_element(self,
+                   element: object,
+                   timestamp: int,
+                   window: CountWindow,
+                   ctx: Trigger.TriggerContext) -> TriggerResult:
+        state = ctx.get_partitioned_state(self._count_state_descriptor)  # type: ReducingState
+        state.add(1)
+        if state.get() >= self._window_size:
+            # On FIRE, the window is evaluated and results are emitted. The window is not purged
+            #               though, all elements are retained.
+            return TriggerResult.FIRE
+        else:
+            # No action is taken on the window.
+            return TriggerResult.CONTINUE
+
+    def on_processing_time(self,
+                           time: int,
+                           window: CountWindow,
+                           ctx: Trigger.TriggerContext) -> TriggerResult:
+        # No action is taken on the window.
+        return TriggerResult.CONTINUE
+
+    def on_event_time(self,
+                      time: int,
+                      window: CountWindow,
+                      ctx: Trigger.TriggerContext) -> TriggerResult:
+        # No action is taken on the window.
+        return TriggerResult.CONTINUE
+
+    def on_merge(self, window: CountWindow, ctx: Trigger.OnMergeContext) -> None:
+        ctx.merge_partitioned_state(self._count_state_descriptor)
+
+    def clear(self, window: CountWindow, ctx: Trigger.TriggerContext) -> None:
+        state = ctx.get_partitioned_state(self._count_state_descriptor)
+        state.clear()
+
+
+class TumblingWindowAssigner(WindowAssigner[object, TimeWindow]):
+    """
+    A WindowAssigner that windows elements into windows based on the current system time of the machine the operation is running on. Windows cannot overlap.
+    For example, in order to window into windows of 1 minute, every 10 seconds:
+    ::
+            >>> data_stream.key_by(lambda x: x[0], key_type=Types.STRING()) \
+            >>> .window(TumblingWindowAssigner(Time.minutes(1), Time.seconds(10), False))
+
+    A WindowAssigner that windows elements into windows based on the timestamp of the elements. Windows cannot overlap.
+    For example, in order to window into windows of 1 minute:
+     ::
+            >>> data_stream.assign_timestamps_and_watermarks(watermark_strategy) \
+            >>> .key_by(lambda x: x[0], key_type=Types.STRING()) \
+            >>> .window(TumblingWindowAssigner(Time.minutes(1), Time.seconds(0), True))
+    """
+    def __init__(self, size: Time, offset: Time, is_event_time: bool):
```

Comment:
```suggestion
    def __init__(self, size: int, offset: int, is_event_time: bool):
```
Use int to keep consistent with the Java API. Besides, it would be great to add a few static utility methods just as did in the Java API.